### ISSUE TITLE
Exhausted connection pool on Select with transaction.

### ISSUE THREAD

**Issue by ykrkn** [davidmoten/rxjava2-jdbc#46]
Versions are 0.2.4, 0.2.5-SNAPSHOT

I wrote simple service to pumping data from databases and that solution requires to runs select with transactions (**autoCommit = false**)

It seems transacted connections for Selects not closing correctly and this behavior going to exhausing of the pool. 

My shorter code snippet below 

```
SelectBuilder sb = db.select("select * from test limit 1").fetchSize(1024);
TransactedSelectBuilder txb = sb.transactedValuesOnly();

Flowable<ExecuteQueryResponseRow> flowableResult = txb
        .get((rs) -> create(rs))
        .map(tx -> tx.value());
return Flux.from(flowableResult);
```

I did try to use nonBlockingPool and Database.fromBlocking with the same result.
In my opinion the reason is **org.davidmoten.rx.jdbc.TransactedSelectBuilder**
I did set **maxPoolSize = 1** to replay the case. Below the logs to explain the problem. 
Database was PostreSQL9.x but i guess this does not matter.
```
-- prepare select and gets the result
13:04:51.171 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.SqlInfo - sqlAfterSubs=select * from test limit 1
13:04:51.199 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - creating new TransactedConnection
13:04:51.200 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - constructing TransactedConnection from org.davidmoten.rx.jdbc.pool.internal.PooledConnection@d7f94f2, 1
13:04:51.200 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - Select.create called with con=TransactedConnection [con=org.davidmoten.rx.jdbc.pool.internal.PooledConnection@d7f94f2, counter=1]
13:04:51.203 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.SqlInfo - sqlAfterSubs=select * from test limit 1
13:04:51.203 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - preparing statement: select * from test limit 1
13:04:51.209 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - parameters=[]
13:04:51.209 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - names=[]
13:04:51.224 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - getting row from ps=org.davidmoten.rx.jdbc.TransactedPreparedStatement@47ab9d1a, rs=org.davidmoten.rx.jdbc.TransactedResultSet@75631083
13:04:51.226 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - emitting ExecuteQueryResponseRow(columns=null, data=[2000-01-01 00:00:00.0])
13:04:51.271 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - getting row from ps=org.davidmoten.rx.jdbc.TransactedPreparedStatement@47ab9d1a, rs=org.davidmoten.rx.jdbc.TransactedResultSet@75631083
-- tx completed
13:04:51.271 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - completed
-- close the statement
13:04:51.271 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing org.davidmoten.rx.jdbc.TransactedPreparedStatement@47ab9d1a
-- closing connection will be skipped because the transaction was not commited yet
13:04:51.271 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing TransactedConnection [con=org.davidmoten.rx.jdbc.pool.internal.PooledConnection@d7f94f2, counter=1]
13:04:51.271 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection attempt close
13:04:51.271 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing org.davidmoten.rx.jdbc.TransactedResultSet@75631083
-- commit the tx but connection will not be closed anymore
13:04:51.273 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection commit attempt, counter=1
13:04:51.273 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection actual commit
-- next select waits the connect eternally
13:04:55.856 [reactor-http-nio-3] DEBUG org.davidmoten.rx.jdbc.SqlInfo - sqlAfterSubs=select * from test limit 1
```

Ok, i watch a codebase.

**org.davidmoten.rx.jdbc.TransactedUpdateBuilder** 
seems ok and connection closing after the tx commited.

```
.doOnNext(tx -> {
    t[0] = ((TxImpl<Integer>) tx);
}) //
.doOnComplete(() -> {
    TxImpl<?> tx = t[0];
    if (tx.isComplete()) {
        tx.connection().commit();
    }
    Util.closeSilently(tx.connection());
}));
```

**org.davidmoten.rx.jdbc.TransactedSelectBuilder**
But here i no see something like that. Hmmm, this is the bug or i do something wrong? 

```
.doOnNext(tx -> {
    if (tx.isComplete()) {
        ((TxImpl<T>) tx).connection().commit();
    }
});

```
After adding the line all works as expected.

```
.doOnNext(tx -> {
    if (tx.isComplete()) {
        ((TxImpl<T>) tx).connection().commit();
        ((TxImpl<T>) tx).connection().close();
    }
});
```

The log for the fixed code.

```
13:09:26.801 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.SqlInfo - sqlAfterSubs=select * from test limit 1
-- omit irrelevant details
-- tx completed
13:09:26.894 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Select - completed
13:09:26.894 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing org.davidmoten.rx.jdbc.TransactedPreparedStatement@1ca97e02
-- closing connection will be skipped because the transaction was not commited yet
13:09:26.894 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing TransactedConnection [con=org.davidmoten.rx.jdbc.pool.internal.PooledConnection@6d78bf70, counter=1]
13:09:26.895 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection attempt close
13:09:26.895 [reactor-http-nio-2] DEBUG org.davidmoten.rx.jdbc.Util - closing org.davidmoten.rx.jdbc.TransactedResultSet@2bdbe545
-- commit the tx
13:09:26.896 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection commit attempt, counter=1
13:09:26.896 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection actual commit
-- And finally successfully close the connection 
13:09:26.897 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection attempt close
13:09:26.897 [reactor-http-nio-2] DEBUG o.d.rx.jdbc.TransactedConnection - TransactedConnection close
```

**Comment by davidmoten** [comment]
Hmm, this slipped my attention very convincingly. I'll see if I can set up a failing test case.

**Comment by davidmoten** [comment]
My fervent apologies for such a long delay. Thanks for making this easy @ykrkn! This is fixed in release 0.2.10 which is on Maven Central now.
